### DISCUSSION TITLE
Cannot initiate batchsensor with irradiancemeter

### DISCUSSION THREAD

**Discussion by cocoakang** [mitsuba-renderer/mitsuba3#880] [Q&A] [answered]
Hi all, 

I want to use a group of irradiancemeters to measure a scene. Iteratively render each meter is time consuming, so I tried to use batch sensor to wrap the irradiancemeters and render the results. However, it seems the batch sensor doesn't think the irradiancemeter as a sensor. I guess because it is wrapped in a Shape object? An example code to reproduce the problem is as below:

```
import mitsuba as mi
mi.set_variant("cuda_ad_rgb")
  
sensor_dict = {
    "type":"batch",
    "meter_0":{
        "type": "rectangle",
        "sensor": {
            "type": "irradiancemeter",
            "film_0": {
                "type": "hdrfilm",
                "width": 1,
                "height": 1,
                "filter": { "type": "box" }
            }
        },
    }
}

scene = mi.load_dict(sensor_dict)
```

It will produce this output:

```
Traceback (most recent call last):
  File "/data/volume1/cocoakang/k.kang/OCT24/cyclops_ir/playground.py", line 20, in <module>
    scene = mi.load_dict(sensor_dict)
RuntimeError: ​[BatchSensor] BatchSensor: at least one child sensor must be specified!
```

I am wondering if I use the batch sensor in a wrong way. If this feature has not been supported yet, is there any other way to simultaneously measure irradiance of a group of positions? Many thanks in advance!

**Comment by leroyvn** [comment, marked as answer]
Hi @cocoakang, I'm afraid you're touching on a limitation of this sensor plugin: it is designed to be declared as a child node to the `Shape` defining the surface on which the incident flux is to be computed; `batch` expects `Sensor` child nodes so it won't accept a `rectangle` (`Shape`) wrapping an `irradiancemeter`.

I think this could be addressed by modifying the plugin to accept a `shape` as an (optional) parameter, opening it to new use cases in the process.

**Reply by cocoakang** [reply]
Thank you for the quick reply! I will try to modify the code. :)

**Reply by njroussel** [reply]
:+1:  I've added this functionnality in https://github.com/mitsuba-renderer/mitsuba3/pull/881/commits/68b8e16edada08c612c950a63b01c3b32aab6d02 (will be merged onto `master` soon, currently in https://github.com/mitsuba-renderer/mitsuba3/pull/881)
